views/006_config_views.sql

```diff
@@ -610,24 +518,50 @@ BEGIN
       configs.created_at,
       configs.updated_at,
       configs.agent_id
-    FROM related_config_ids_recursive($1, $2, $3) as r
-    LEFT JOIN configs ON r.id = configs.id;
+   FROM (
+   	SELECT
+   	  r.id::uuid,
+   	  min(r.related_id::text)::uuid as related_id,
+   	  min(r.relation_type) as relation_type,
+      min(r.direction) as direction,
+      min(r.depth) as depth
+    FROM related_config_ids_recursive($1, $2, $4) as r
+    GROUP BY r.id
+   ) r
+    LEFT JOIN configs ON r.id = configs.id
+    WHERE related_configs_recursive.include_deleted_configs OR configs.deleted_at IS NULL
+    ORDER BY depth ASC;
 END;
 $$ LANGUAGE plpgsql;
 
--- related config changes recursively
-DROP FUNCTION IF EXISTS related_changes_recursive(UUID, TEXT, BOOLEAN);
+-- related configs
+DROP FUNCTION IF EXISTS related_configs;
 
-CREATE FUNCTION related_changes_recursive (
```

Comment:
@moshloop Why are we removing this? It's being used by the catalog/changes endpoint
https://github.com/flanksource/duty/blob/fb9b1e136d24775cff357caa92a6d7a7370fe551/query/config_changes.go#L142